src/git-http/sync.ml

```diff
@@ -454,23 +417,28 @@ module Make_ext
             (fun _ -> Lwt.return ())
         in
 
-        let x, r =
-          List.map (function
-              | `Create (hash, reference) -> Encoder.Create (hash, Store.Reference.to_string reference)
-              | `Delete (hash, reference) -> Encoder.Delete (hash, Store.Reference.to_string reference)
-              | `Update (_of, _to, reference) -> Encoder.Update (_of, _to, Store.Reference.to_string reference))
-            commands
-          |> fun commands -> List.hd commands, List.tl commands
-        in
-
-        packer ~window:(`Object 10) ~depth:50 ~ofs_delta:true git refs.Decoder.refs (x :: r) >>= function
+        Lwt_list.map_p (function
```

Comment:
why do you use `map_p` here? There is no blocking operation so everything will happen sequentially anyway.